src/pal/src/locale/utf8.cpp

```diff
@@ -2022,7 +2022,7 @@ class UTF8Encoding
 
                         ch = (chc << 6) | (ch & 0x3F);
 
-                        *pTarget = (char)(((ch >> 10) & 0x7FF) +
+                        *pTarget = (WCHAR)(((ch >> 10) & 0x7FF) +
                             (SHORT)(CharUnicodeInfo::HIGH_SURROGATE_START - (0x10000 >> 10)));
```

Comment:
Should the `(SHORT)` also be updated to `(WCHAR)`?

Reply:
For this case, there two are same. And this is a services fix, so I try to change as less as possible. :)

Reply:
I can appreciate they are presently the same, but the concern I have is related to the UTF-8 work that is happening in another PR. Having these subtle assumptions in the codebase is a recipe for bugs like the one being fixed here. Ensuring the character size/type is consistent across the codebase should be a priority. Other VMs tend to have what is called a 'rune' that represents a character that hides the current assumed encoding - WCHAR seems to serve that purpose in the coreclr repo this case.